From: Joe Shaw <joeshaw@novell.com>

Optimize MultiTermDocs.SkipTo() to be much faster in the case of indexes which
are contain multiple segments (ie, unoptimized).  This helps search
performance quite a bit on unoptimized indexes.  This is a backport from the
Lucene devel 2.1 branch.

See https://issues.apache.org/jira/browse/LUCENE-586

Index: Index/MultiReader.cs
===================================================================
RCS file: /cvs/gnome/beagle/beagled/Lucene.Net/Index/MultiReader.cs,v
retrieving revision 1.3
diff -u -p -u -r1.3 MultiReader.cs
--- Index/MultiReader.cs	2 Oct 2006 17:09:00 -0000	1.3
+++ Index/MultiReader.cs	16 Oct 2006 17:36:37 -0000
@@ -535,16 +535,16 @@ namespace Lucene.Net.Index
 			}
 		}
 		
-		/// <summary>As yet unoptimized implementation. </summary>
 		public virtual bool SkipTo(int target)
 		{
-			do 
-			{
-				if (!Next())
-					return false;
-			}
-			while (target > Doc());
-			return true;
+			if (current != null && current.SkipTo (target - base_Renamed)) {
+				return true;
+			} else if (pointer < readers.Length) {
+				base_Renamed = starts [pointer];
+				current = TermDocs (pointer++);
+				return SkipTo (target);
+			} else
+				return false;
 		}
 		
 		private TermDocs TermDocs(int i)
